src/Engine/Utils.java

```diff
@@ -501,6 +501,63 @@ public static void sortLocationsByTravelTime(Unit unit, ArrayList<XYCoord> mapLo
     TravelDistanceComparator tdc = new TravelDistanceComparator(unit, map);
     Collections.sort(mapLocations, tdc);
   }
+  
+
+  /** Returns a string describing the game statistics of a given CO */
+  public static String getGameStatusData(GameMap map, Commander viewed)
+  {
+    StringBuilder sb = new StringBuilder();
+
+    // map state
+    int income = 0;
+    int unitCount = 0;
+    int vehCount = 0;
+    int unitFunds = 0;
+    
+    for( int w = 0; w < map.mapWidth; ++w )
+    {
+      for( int h = 0; h < map.mapHeight; ++h )
+      {
+        Location loc = map.getLocation(w, h);
+        if( loc.isProfitable() && loc.getOwner() == viewed )
+        {
+          income += viewed.incomePerCity;
+        }
+        Unit resident = loc.getResident();
+        if( null != resident && resident.CO == viewed )
+        {
+          unitCount++;
+          if (resident.model.chassis != ChassisEnum.TROOP)
+            vehCount++;
+          unitFunds += resident.model.getCost() * resident.getHP() / resident.model.maxHP;
+        }
+      }
+    }
+
+    sb.append("Income:         ").append(income).append("\n");
+    sb.append("Unit count:     ").append(unitCount).append("\n");
+    sb.append("Vehicle count:  ").append(vehCount).append("\n");
```

Comment:
Are we gonna split out air and sea units next too?

Reply:
Nah. People didn't ask for that. These categories were specifically requested.